### PR TITLE
SA1509 Opening braces should not be preceded by blank line

### PR DESCRIPTION
**Pull request by elachlan**
Relates to #7174
https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1509.md

### CODE REVIEW

**Review by rokonec — APPROVED** [review]

File: src/Tasks.UnitTests/XslTransformation_Tests.cs [hunk 1]
```diff
@@ -864,7 +864,6 @@ public void CompiledDllWithTwoTypes()
             string doubleTypePath = Path.Combine(dir, "double.dll");
 
             CompileDoubleType(doubleTypePath);
-
```
Comment (ladipro) [inline src/Tasks.UnitTests/XslTransformation_Tests.cs:867]: When the braces are used only as a code block - so no class, method body, conditional, loop, etc. - I actually think that an empty line can help. Is it possible that the rule is not working as intended?

Comment (ladipro) [reply src/Tasks.UnitTests/XslTransformation_Tests.cs:867]: cc @sharwell

Comment (elachlan) [reply src/Tasks.UnitTests/XslTransformation_Tests.cs:867]: I see what you mean, but I don't understand why anyone would add braces around code like this? Are we able to just remove them?

Comment (ladipro) [reply src/Tasks.UnitTests/XslTransformation_Tests.cs:867]: There are some niche uses for code blocks. It's nice to be able to reuse variable names sometimes:

```C#
{
    int x = 42;
    // ...
}
{
    string x = "hello";
    // ...
}
```

And the scope applies to `using` also, so I can imagine favoring code like this:

```C#
{
    using var x = new Resource();
    using var y = new Resource();
    using var z = new Resource();
    // ...
}
```

over:

```C#
using (var x = new Resource())
{
    using (var y = new Resource())
    {
        using (var z = new Resource())
        {
          // ...
        }
    }
}
```

But I guess it's generally avoidable. In this case I would vote for just removing it.

Comment (rokonec) [reply src/Tasks.UnitTests/XslTransformation_Tests.cs:867]: I have also seen code blocks used in ifdef scenarios:
```C#
#if CONDITIONAL_THINGY
  if (this || that)
#endif
  {
    Thingy1();
    Thingy2();
  }
```
But in this particular case I also vote for removing the code block.

**Review by AR-May — APPROVED** [review]

File: src/Tasks.UnitTests/XslTransformation_Tests.cs (hunk 1 above)
Comment (elachlan) [reply src/Tasks.UnitTests/XslTransformation_Tests.cs:867]: @rokonec in your example the braces don't have a blank line before hand. So it wouldn't trigger the rule.

This rule is about a blank line before open braces.

**Review by Forgind — APPROVED** [review]

**Pull request merged**